### ISSUE TITLE
3.0.0-RC1 Change Checkbox Template 

### ISSUE THREAD

**Issue by basteyy** [cakephp/cakephp#5708]
I want to change the ordner of my checkbox and the label.

I tryed to change the checkboxFormGroup. But the template dont know the {{iput}}. Its not possible to change the order. 

Version: 3.0.0-RC1

**Comment by markstory** [comment]
The ['checkboxFormGroup' does get the input as a parameter](https://github.com/cakephp/cakephp/blob/3.0/src/View/Helper/FormHelper.php#L1023-L1031).

How did you try changing the template?

**Comment by basteyy** [comment]
First via an custom config file, than direct in the FormHelper. 

See here: https://gist.github.com/basteyy/d0ec08a2355c7b4863fc

**Comment by AD7six** [comment]
@basteyy why use a gist instead of putting that in the ticket description?

Seems like the config in your case is invalid, if you look at [the line generating the warning](https://github.com/cakephp/cakephp/blob/3.0/src/View/StringTemplate.php#L216) what is `$template` ?

**Comment by basteyy** [comment]
Dont know how to use the markdown syntax support. Sorry. 
It looks quite good for me. Same value like any other form element. 
![cake](https://cloud.githubusercontent.com/assets/2758879/5841268/ed30751c-a19b-11e4-94dc-54afae046c6b.PNG)

**Comment by AD7six** [comment]
You are debugging something different that the ticket description @basteyy - identify the source of this:

    Array to string conversion [CORE\src\View\StringTemplate.php, line 21

That, or better explain the problem because right now it looks like user error/wrong usage.

**Comment by basteyy** [comment]
I try again: I want to change the template for checkboxes which are provided by the form helper. I want the label first and then the input-field (the checkbox). Seems that {{label}} contains the <label> and the <input type="checkbox">

I debugged the CORE\src\View\StringTemplate.php, line 216 when the script create the checkbox-input. And that is what you see on the image. So what else should I debug? 

Can you change the order of showing elements in a checkbox area?

**Comment by Xety** [comment]
Try this in your `AppController.php`:
```php
public $helpers = [
	'Form' => [
		'templates' => [
			'checkboxFormGroup' => '{{label}}{{input}}'
		]
	]
];
```
http://book.cakephp.org/3.0/en/views/helpers/form.html#list-of-templates

**Comment by basteyy** [comment]
The problem is not defining the template. I use an extra config file (loaded via View/AppView.php). 
Moreover defining it in the AppController produced the same error. Dies anyone have the same problem or do you all can easily use {{input}} in the checkboxFormGroup-Template?

**Comment by AD7six** [comment]
The method format is called many times - In the image above $name is "checkbox" not "checkboxFormGroup". All you need to do is look at the stack trace for the notice _and put it in the ticket description_. It's currently unclear what code you've got, what output you're getting and how to reproduce your scenario.

However if I do this:

```
-> cat config/app_form.php 
<?php
$config = [
    'checkboxFormGroup' => '{{label}} {{input}}'
];
www-data @ dev [ /var/www/cakephp.dev ] (master *$%=)
-> cat src/Template/Pages/home.ctp 
START

<?php echo $this->Form->input('foo', ['type' => 'checkbox']); ?>

END
```

The output is this:

```
-> php webroot/index.php 
<!DOCTYPE html>
...
                START
<div class="input checkbox"><label for="foo">Foo</label> <input type="hidden" name="foo" value="0"><input type="checkbox" name="foo" value="1" id="foo"></div
END
...
</html>
```

Which has the label before the input and appears to be what you're trying to do.

Closing as there appears to be nothing wrong.

**Comment by Xety** [comment]
Okay, just tested and he's right :
![screenshot_2](https://cloud.githubusercontent.com/assets/8210023/5844156/1b6f8fe2-a1b2-11e4-9efa-ffd3544cc235.png)
![screenshot_3](https://cloud.githubusercontent.com/assets/8210023/5844162/259bc454-a1b2-11e4-8713-235c68b427cd.png)
![screenshot_4](https://cloud.githubusercontent.com/assets/8210023/5844166/32d738e2-a1b2-11e4-8fa6-1c32f2f20fe3.png)

**Comment by basteyy** [comment]
I use windows (like Xety). Perhaps the reason?

**Comment by AD7six** [comment]
@Xety thanks for providing something specific to look at. 

It's not a problem of windows, but it's already fixed in the 3.0 branch (I now know, as I can see the same error using 3.0.0-RC instead of the current 3.0 branch).

So, the problem when it occurs is that `replace` is a 2d array:

```
/src/View/StringTemplate.php (line 217)
########## DEBUG ##########
[
	(int) 0 => '<input type="hidden" name="foo" value="0"><label for="foo"><input type="checkbox" name="foo" value="1" id="foo">Foo</label>',
	(int) 1 => [
		'hidden' => '<input type="hidden" name="foo" value="0">',
		'input' => '<input type="checkbox" name="foo" value="1" id="foo">'
	]
]
###########################
```

Leaving closed as the fix (hard to find amongst the psr changes) has already been applied.

**Comment by basteyy** [comment]
I have updated to RC-2 and still the problem. Is there no solution?
